src/sync/__tests__/syncManagerOnline.spec.ts

```diff
@@ -57,41 +97,99 @@ test('syncManagerOnline should start or not the submitter depending on user cons
 test('syncManagerOnline should syncAll a single time in singleSync mode', () => {
   const settings = { ...fullSettings };
 
+  // Enable single sync
   settings.sync.singleSync = true;
 
-  const pollingManager = {
-    syncAll: jest.fn(),
-    start: jest.fn(),
-    stop: jest.fn(),
-    isRunning: jest.fn()
-  };
+  // @ts-ignore
+  const pollingSyncManager = syncManagerOnlineFactory(() => pollingManagerMock)({ settings });
```

Comment:
I see that we are duplicating same asserts with 2 syncManager instances: `const pollingSyncManager = syncManagerOnlineFactory(() => pollingManagerMock)({ settings });` and `const pushingSyncManager = syncManagerOnlineFactory(() => pollingManagerMock, () => pushManagerMock)({ settings });`

I would have only one:
`const syncManager = syncManagerOnlineFactory(() => pollingManagerMock, pushManagerFactoryMock)({ settings });`

where `pushManagerFactoryMock` is a `Jest.fn` mock you can use to assert that the pushManager is not created when `singleSync` is false